client/my-sites/earn/components/add-edit-coupon-modal/test/products-selector.jsx

```diff
@@ -0,0 +1,210 @@
+	type: 'newsletter',
+	is_editable: false,
+};
+
+const initialState = {
+	sites: { items: {} },
+	siteSettings: { items: {} },
+	ui: { selectedSiteId: 1 },
+	memberships: {
+		couponList: {
+			items: {},
+		},
+		productList: {
+			items: {
+				1: [ testProduct1, testProduct2 ],
+			},
+		},
+	},
+};
+
+function WrappedProductsSelector( props ) {
+	const mockStore = configureStore();
+	const store = mockStore( initialState );
+
+	return (
+		<Provider store={ store }>
+			<ProductsSelector { ...props } />
+		</Provider>
+	);
+}
+
+describe( 'ProductsSelector', () => {
+	const getAnyProductCheckbox = () =>
+		screen.getByRole( 'menuitemcheckbox', { name: 'Any product' } );
+	const getYearlyProductCheckbox = () =>
+		screen.getByRole( 'menuitemcheckbox', {
+			name: 'Yearly Subscription : $40.00 / year',
+		} );
+	const getMonthlyProductCheckbox = () =>
+		screen.getByRole( 'menuitemcheckbox', {
+			name: 'Monthly Subscription : $5.00 / month',
+		} );
+	const getButtonText = ( selectedCount ) => {
+		if ( selectedCount > 0 ) {
+			if ( selectedCount > 1 ) {
+				return selectedCount + ' products selected.';
+			}
+			return '1 product selected';
+		}
+		return 'Any product';
+	};
+	const getDropdownButton = ( selectedCount ) =>
+		screen.getByRole( 'button', { name: getButtonText( selectedCount ) } );
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+
+		useTranslate.mockImplementation( mockUseTranslate );
+		translate.mockImplementation( mockTranslate );
+		__.mockImplementation( mockTranslate );
+	} );
+
+	test( 'should render with any product selected when no specific products have been selected', () => {
+		let editedPlanIdsAllowList = [];
+		const setEditedPlanIdsAllowList = ( list ) => {
+			editedPlanIdsAllowList = list;
+		};
+		render(
+			<WrappedProductsSelector
+				onSelectedPlanIdsChange={ ( list ) => setEditedPlanIdsAllowList( list ) }
+				initialSelectedList={ editedPlanIdsAllowList }
+				allowMultiple={ true }
+			/>
+		);
+
+		act( () => getDropdownButton( editedPlanIdsAllowList.length ).click() );
+		const anyProductCheckbox = getAnyProductCheckbox();
+		const yearlyProductCheckbox = getYearlyProductCheckbox();
+		const monthlyProductCheckbox = getMonthlyProductCheckbox();
+
+		expect( anyProductCheckbox ).toBeChecked();
+		expect( yearlyProductCheckbox ).toBeChecked();
+		expect( monthlyProductCheckbox ).toBeChecked();
+	} );
+
+	test( 'should render with 1 product selected when 1 product has been selected', () => {
+		let editedPlanIdsAllowList = [ testProduct1.ID ];
+		const setEditedPlanIdsAllowList = ( list ) => {
+			editedPlanIdsAllowList = list;
+		};
+		render(
+			<WrappedProductsSelector
+				onSelectedPlanIdsChange={ ( list ) => setEditedPlanIdsAllowList( list ) }
+				initialSelectedList={ editedPlanIdsAllowList }
+				allowMultiple={ true }
+			/>
+		);
+		// Expect the button with the correct name to be rendered.
+		expect( getDropdownButton( editedPlanIdsAllowList.length ) ).not.toBeNull();
```

Comment:
Can we make this more precise? What value precisely do we expect? Same for a few more similar assertions in the same file.

Reply:
Not really, and maybe that's a clue to an accessibility problem with this component. This is a multiple "select" component that is a Button component to begin with and when clicked reveals a menu. Maybe it would be better to refactor this component to more closely mimic a select dropdown.

For now I've changed this to `.toBeInTheDocument` since the assertion is mostly checking to make sure that the correct _number_ of products have been selected (but not that the correct products are selected).